src/main/java/org/broadinstitute/barclay/argparser/TaggedArgumentParser.java

```diff
@@ -0,0 +1,333 @@
+     *
+     * @param taggedArg implementation of TaggedArgument interface
+     * @return a display string representing the tag name and attributes
+     */
+    public static String getDisplayString(final TaggedArgument taggedArg) {
+        Utils.nonNull(taggedArg);
+
+        StringBuilder sb = new StringBuilder();
+        sb.append(ARGUMENT_TAG_NAME_SEPARATOR);
+        sb.append(taggedArg.getTag() == null ? "NoTagName" : taggedArg.getTag());
+
+        // check for nulls since we can't assume the implementation is compliant
+        if (taggedArg.getTagAttributes() != null) {
+            taggedArg.getTagAttributes().entrySet().stream()
+                    .filter(entry -> entry.getValue() != null)
+                    .forEach(
+                            entry -> {
+                                sb.append(ARGUMENT_KEY_VALUE_PAIR_DELIMITER);
+                                sb.append(entry.getKey().toString());
+                                sb.append(ARGUMENT_KEY_VALUE_SEPARATOR);
+                                sb.append(entry.getValue().toString());
+                            });
+        }
+        return sb.toString();
+    }
+
+    /**
+     * Represents a parsed, tagged argument.
+     *
+     * May have attributes.
+     */
+    private static final class ParsedArgument{
+        private final String name;
+        private final Map<String, String> keyValueMap;
+
+        /**
+         * Parses an argument value String of the forms:
+         *
+         * "logical_name(,key=value)"
+         *
+         * into logical name and key=value pairs.
+         *
+         * @param rawTagValue argument value from the command line to parse
+         * @return The argument parsed from the provided string.
+         */
+        public static ParsedArgument of(final String rawTagValue) {
+            final String[] tokens = rawTagValue.split(ARGUMENT_KEY_VALUE_PAIR_DELIMITER);
+            if ( Arrays.stream(tokens).anyMatch(String::isEmpty)) {
+                throw new CommandLineException.BadArgumentValue("", rawTagValue, "Empty tag name encountered. " + USAGE);
+            }
+            if (tokens.length == 0) {
+                throw new CommandLineException.BadArgumentValue("", rawTagValue, USAGE);
+            }
+
+            // first token is required to be a name
+            if (tokens[0].contains(ARGUMENT_KEY_VALUE_SEPARATOR)) {
+                throw new CommandLineException.BadArgumentValue("Missing name for argument: " + rawTagValue);
+            }
+            final ParsedArgument pa = new ParsedArgument(tokens[0]);
+
+            if (tokens.length == 1) {
+                // even though its parseable, notify the user for the case where an attribute is indicated, but not
+                // included, such as for a dangling comma ("--t:tumor,") since it may be user error
+                if (rawTagValue.contains(ARGUMENT_KEY_VALUE_PAIR_DELIMITER)) {
```

Comment:
If you call split with `-1` above, you should be able to delete this check, since split will not drop the empty tokens and they'll be caught by the `isEmpty` check above.